snuba/migrations/snuba_migrations/metrics/0024_metrics_distributions_add_histogram.py

```diff
@@ -0,0 +1,81 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import AggregateFunction, Column, Float
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.snuba_migrations.metrics.templates import (
+    COL_SCHEMA_DISTRIBUTIONS_V2,
+    get_forward_view_migration_polymorphic_table_v2,
+    get_polymorphic_mv_v3_name,
+)
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Add HdrHistogram to distribution metrics.
+    """
+
+    blocking = False
+    raw_table_name = "metrics_raw_local"
+
+    def __forward_migrations(
+        self, table_name: str
+    ) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.METRICS,
+                table_name=table_name,
+                column=Column(
+                    "histogram_buckets",
+                    AggregateFunction("histogram(250)", [Float(64)]),
+                ),
+                after="count",
+            )
+        ]
+
+    def __backward_migrations(
+        self, table_name: str
+    ) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.DropColumn(
+                storage_set=StorageSetKey.METRICS,
+                table_name=table_name,
+                column_name="histogram_buckets",
+            )
+        ]
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            *self.__forward_migrations("metrics_distributions_local"),
+            get_forward_view_migration_polymorphic_table_v2(
```

Comment:
@onewland pointed out that bumping the enabled materialization version will stop writing into the aggregation tables for everything _but_ distributions. We need to re-create all materialized views for metrics here.

Reply:
yeah, it's a bit of a pain but that's probably the cleanest approach

Reply:
We should not need this. 
So, if you want to recreate all the matviews, there has to be a migration with the three. 
Like this https://github.com/getsentry/snuba/pull/2524/files#diff-255f09f391ed726eb0533354f3dc4836bb4c1d66958fb1b06a0e118788fc15c0R1

In alternative, you can change the consumer and write a different version per metric type and only create a new matview for distribution that has a different version from the others.

Considering we are still not in prod and making change every day, up to you. The only critical element is that the migrations need to stay consistent and that you need to be able to achieve a stable state when running all with no manual intervention.